Source/utils/soundsample.cpp

```diff
@@ -16,90 +21,77 @@ namespace devilution {
 
 void SoundSample::Release()
 {
-	Mix_FreeChunk(chunk);
-	chunk = nullptr;
+	stream_ = std::nullopt;
+	file_data_ = nullptr;
 };
 
 /**
  * @brief Check if a the sound is being played atm
  */
 bool SoundSample::IsPlaying()
 {
-	if (chunk == nullptr)
-		return false;
-
-	int channels = Mix_AllocateChannels(-1);
-	for (int i = 0; i < channels; i++) {
-		if (Mix_GetChunk(i) == chunk && Mix_Playing(i) != 0) {
-			return true;
-		}
-	}
-
-	return false;
+	return stream_.has_value() && stream_->isPlaying();
 };
 
 /**
  * @brief Start playing the sound
  */
 void SoundSample::Play(int lVolume, int lPan, int channel)
 {
-	if (chunk == nullptr)
+	if (!stream_.has_value())
 		return;
 
-	channel = Mix_PlayChannel(channel, chunk, 0);
-	if (channel == -1) {
-		SDL_Log("Too few channels, skipping sound");
-		return;
+	stream_->setVolume(static_cast<float>(std::pow(10.0, lVolume / 2000.0)));
+	if (lPan == 0) {
+		stream_->setStereoPosition(0);
+	} else {
+		stream_->setStereoPosition(
+		    std::copysign(
+		        std::pow(10.F, static_cast<float>(-std::fabs(lPan) / 2000.F)),
+		        static_cast<float>(lPan)));

```

Comment:
There's a '1 - ' missing, otherwise it will actually pan samples towards the center the further to the left/right the sound source is.

```suggestion
		    std::copysign(
		        1.0f - std::pow(10.F, static_cast<float>(-std::fabs(lPan) / 2000.F)),
		        static_cast<float>(lPan)));

```